pkg/networkserver/grpc_asns.go

```diff
@@ -310,7 +319,10 @@ func (ns *NetworkServer) DownlinkQueueReplace(ctx context.Context, req *ttnpb.Do
 
 // DownlinkQueuePush is called by the Application Server to push a downlink to queue for a device.
 func (ns *NetworkServer) DownlinkQueuePush(ctx context.Context, req *ttnpb.DownlinkQueueRequest) (*pbtypes.Empty, error) {
-	if len(req.Downlinks) == 0 {
+	switch n := len(req.Downlinks); {
+	case uint64(n) > ns.downlinkQueueCapacity:
+		return nil, errDownlinkQueueCapacityExceeded.New()
+	case n == 0:
 		return ttnpb.Empty, nil
 	}
```

Comment:
Another more readable version that has less of a show off factor of what you can do with switch/case:

```suggestion
	if uint64(len(req.Downlinks)) > ns.downlinkQueueCapacity {
		return nil, errDownlinkQueueCapacityExceeded.New()
	}
	if len(req.Downlinks) == 0 {
		return ttnpb.Empty, nil
	}
```

Note that `len()` is not counting anything

Reply:
I don't care about showing off, what I do care about is readability and maintainability.
In my opinion `switch` is the most readable option when there are 2 or more cases.
IMO `else if` structure should be used only in very specific cases.

I refactored into `else if`